physics/module_mp_thompson.F90

```diff
@@ -2234,7 +2252,7 @@ subroutine mp_thompson (qv1d, qc1d, qi1d, qr1d, qs1d, qg1d, ni1d,    &
             if (xDi.lt. 5.E-6) then
              lami = cie(2)/5.E-6
              ni(k) = MIN(4999.D3, cig(1)*oig2*ri(k)/am_i*lami**bm_i)
-            elseif (xDi.gt. 300.E-6) then
+            elseif (xDi.gt. D0s + 100.E-6) then
```

Comment:
This is incorrect.  The very next line has a hard-wired value of 300 (as it should). There is a lack of understanding to the root point of limiting the number concentration in the mathematics to compute lambda.  This line should be reverted to the previous hard-wired number.  The choice of D0s is not necessarily directly related to the choice of largest allowable mass-weighted mean size of cloud ice.

Reply:
The change was made because when D0s was 200 um before  300 was used here.  I agree that there is no definite number.   If you think 300 um is large enough for D0s = 300 um it is fine to use 300 and revert the change.

Reply:
Reverted as suggested.

Reply:
@RuiyuSun : but you're not getting my point.  The line following the 300 (microns) is the limit of the lambda calculation which is how it gets hard-wired.  So people should not change one line without changing the other, because the mathematics can be *wrong* if someone further changes ```D0s```.  More investigation should take place when changing numbers in code that are not up at the top. I declared them as far up in the code as possible because those are the values most likely to be changed, and I am trying to protect other lines of code from being modified.

Reply:
@gthompsnWRF I've reverted it to 300 as you suggested along with other changes you suggested. Are we good to start the commit process?